### PR TITLE
fix: qnap temperature sensors

### LINKED ISSUE #4586: QNAP Sensors description missing and default thresholds not set,
DO NOT DELETE THIS INFORMATION.

> Please read this information carefully.

- [X] Is your install up to date? [Updating your install](http://docs.librenms.org/General/Updating/)
      Please do not submit an issue if your install is not up to date within the last 24 hours or on a stable monthly release.
- [X] Please include all of the information between the `====================================` section of `./validate.php`.
- [ ] If you would like us to add a new device then please provide the information asked for [here](http://docs.librenms.org/Support/FAQ/#faq20)
- [X] Please provide as much detail as possible.

QNAP snmp sensor description is missing and default thresholds for max and min are 0,
causing alerts to be issued for hdd temps and fan speeds.
HW: QNAP TS-659
OS: QTS-4.2.0 (2016-03-11)

http://pastebin.com/JKjYXzhc

### PR DESCRIPTION
**Pull request by crcro**
#### Please note

> Please read this information carefully. You can run `./scripts/pre-commit.php` to check your code before submitting.

- [x] Have you signed the [Contributors agreement](http://docs.librenms.org/General/Contributing/)
- [x] Have you followed our [code guidelines?](http://docs.librenms.org/Developing/Code-Guidelines/)

fixes: #4586

### CODE REVIEW

File: includes/discovery/sensors/temperatures/qnap.inc.php [hunk 1]
```diff
@@ -23,9 +23,14 @@
             // Get the disk temperature full oid
             $disk_oid = $disk_temperature_oid.$disk_number;
             // Get the temperature for the disk
-            $disk_temperature = $entry['HdTemperature'];
+            $disk_temperature = $entry['hdTemperatureEX'];
```
Comment (laf) [inline includes/discovery/sensors/temperatures/qnap.inc.php]: Are we sure that the old `HdTemperature` isn't version dependant and this won't break it for other users?

Comment (crcro) [inline includes/discovery/sensors/temperatures/qnap.inc.php]: i've put an version check in place to keep the backwards compatibility ...

**Comment by laf** [comment]
Looks good to me.

I'd like to get a second opinion @librenms/reviewers

**Comment by crcro** [comment]
all qnap modifications i've submitted are tested on my qnap

**Comment by laf** [comment]
I understand that but we've been caught out before by updating code to support a newer format and in turn breaking the older device support :(

**Comment by paulgear** [comment]
Per #4590, there are very few qnaps out there according to stats, so if it works on @crcro's qnap that's probably as good an indicator as anything.

**Pull request merged**